### ISSUE TITLE
CRAN release?

### ISSUE THREAD

**Issue by krlmlr** [dirkschumacher/defmacro#9]
Are you planning a CRAN release of this package any time soon?

**Comment by dirkschumacher** [comment]
Do you have a specific usecase? I would not mind putting on CRAN

**Comment by HenrikBengtsson** [comment]
I think I've got plenty of use cases for what you've solved quite elegantly here.  Some examples:

* optional run-time debugging (in the same spirit as **debugme**)
* optional run-time logging, possibly at different amounts (e.g. INFO, DEBUG, TRACE)
* optional run-time assertion, possibly partial or in full during checks (https://github.com/r-lib/debugme/issues/32)
* optional run-time profiling (e.g. enable timing and memory profile of futures for troubleshooting purposes)
* optional progress updates

... all with zero overhead when disabled (=removed on load).  To me, as a developer trying to maximize correctness (=lots of assertions) and performance (=minimal number of asserts), being able to use macros that can be enabled and disabled when the package is loaded is of great value.   Another benefit is that the code that `print(<function>)` outputs is cleaner when above "non-essential" code is disabled/removed.

**Comment by krlmlr** [comment]
I don't mind removing rlang if it causes strong feelings.

Also, I'm ready to tick the "Test the package in a real package :)" bullet. Specifically, I'd like to explore usage of this package for expanding DBItest tests: https://github.com/r-dbi/DBItest/pull/267. The linked PR exports way too many functions from DBItest. If those were macros, the tests would become larger, but also easier to troubleshoot in case of failures.

If a CRAN release in the near future is elusive, adding the package to r-universe would be great too: https://dirkschumacher.r-universe.dev/ui#packages.

**Comment by dirkschumacher** [comment]
> I don't mind removing rlang if it causes strong feelings.
It's already gone from the dependencies.

> Also, I'm ready to tick the "Test the package in a real package :)" bullet. 
Let's do it then. Only thing left is polishing the `DESCRIPTION` file I believe.

**Comment by dirkschumacher** [comment]
BTW. @HenrikBengtsson @krlmlr you both contributed to the package. Happy to add you to the `DESCRIPTION` file. Or better, make a PR :)

**Comment by krlmlr** [comment]
Thanks. I wonder why this package doesn't show up in your R universe. Do you need to set up manually?

**Comment by krlmlr** [comment]
I'll try to start using defmacro in DBItest in the next few days. I need to confirm that it actually solves my use case.

**Comment by dirkschumacher** [comment]
Sure go for it. Or do you need it on cran first?

**Comment by krlmlr** [comment]
No, rather the opposite -- I'd take the opportunity to provide feedback, if necessary, before it hits CRAN.

**Comment by krlmlr** [comment]
It wouldn't hurt if it were on R-universe, though.
